### ISSUE TITLE
Submodules are hard to identify by their titles

### ISSUE THREAD

**Issue by dimitribouniol** [desktop/desktop#1803]
### Description

In GitHub Desktop (both in the old version and the new beta, though I wouldn't be able to tell in the new version due to #708), if you add the same submodule to the sidebar, but from different parent repositories, they will all be listed with the same name. It appears that GitHub Desktop tries to be smart by prefixing it with the GitHub user's username, but this is not entirely useful in figuring out which parent repository they belong to.

### Version

**GitHub Desktop version:** 223 & 0.5.8
**OS version:** macOS 10.12.4

### Steps to Reproduce

1. Add a submodule from parent repo A
1. Add the same submodule from parent repo B

**Expected behaviour:** The repos are listed as such:
```
RepoA -> user/MySubmodule
RepoB -> user/MySubmodule
```
(or something similar)

**Actual behaviour:** The repos are listed as such:
```
user/MySubmodule
user/MySubmodule
```
(which is not very helpful)

Other great differentiators could be the current branch/commit. Ideally, it should be easy to identify which submodule you were working on despite that submodule being everywhere (as submodules are likely to be):
```
RepoA -> org/MySubmodule
RepoB -> user/MySubmodule (bugfix)
RepoB -> user/MySubmodule (master)
user/MySubmodule
```

**Comment by shiftkey** [comment]
Tentatively marking this as enhancement as I think we need to do more work with parsing the status before we can polish the UI. Might also need some design input down the track.

**Comment by chityala123** [comment]
nice job

**Comment by mcmonkey4eva** [comment]
+1 on this but also, it'd be neat to handle this in a general way.
EG two clones of a single repo in different folders should still be differentiated, submodule or not.

**Comment by PAGG-ESkopal** [comment]
+1 I have the same problem.  I am working on two different branches of the same repository, in different directories, so I can build and test them separately.  The name should include some part of the directory name by default.  In all cases, the user should be able to override or append to the name and enter one of their own.

**Comment by dimitribouniol** [comment]
I agree that being able to manually edit the name is a great idea.

**Comment by rsodre** [comment]
This lack of submodule parenting also bothers me a lot, and in my mind it's really easy to solve the UI part of it with a **tree view**.

**Perfect solution**: Any repositories that have submodules should contain a "+" or "►" on the side to unfold and reveal the submodules. They can all be greyed out, and if you check out any of them, it is linked to the github repository, like opening a local repo.

**Acceptable solution**: Even if it does not recognize submodules as I'm suggesting, if I open a local repository that is a submodule of any repository already opened, just parent it below it, indented.

**Comment by velddev** [comment]
I was looking for this issue, and figured to contribute to it with my amateuristic design skills and created a mockup that *seemingly* should solve the issue. Hopefully we can hear an update related to this issue soon 😄 

![image](https://user-images.githubusercontent.com/13852840/51081457-1ca44300-16f0-11e9-9d2d-8ec1dc65637d.png)

**Comment by dimitribouniol** [comment]
Ping @shiftkey, since as others point out, it has been a while 😊

Here is another interpretation how I imagine it can look. In this mockup, I grew the row heights to 43pt, and added another line that contains both the current branch (or truncated commit, if on HEAD), and the local path of the repo, truncated at the beginning of the path if necessary. As I experimented, I realized this information is always useful, so I have it for all rows rather than just the ambiguous paths, as I originally suggested.

The commit is especially useful when comparing to the diff a main repo would show if the submodule was in a different state than what was originally committed. The branch name would also be very useful when switching between different repos that are in different states.

Finally, I would love for this column to always be visible much like the old GitHub desktop, as it facilitated going back and forth between repos very quickly (I'm still using the old app primarily for this reason... sorry 😕). Even better would be to open multiple windows, each showing their own repo, which would provide the best experience for power users.

(I also realize after I uploaded, but the paths should include the folder of the git repo as well, and not just the name of the repo on GitHub, since they can be different. So the first entry would probably read `...inRepo/Libraries/SubMod` if the `.git/` itself is within the directory `SubMod`)

![github-branched-mockup 2x](https://user-images.githubusercontent.com/225505/51082175-e6969d00-16b6-11e9-87cb-e98e5ffd83d9.png)

Some metrics: The row heights grow to 43pt, the icons are shifted down by 7pt, the new text is 10pt, black at 40% opacity. The selected state is white at 75% opacity. The top row of text doesn't change or move in any way. A bullet character `•` is used to separate the branch and the path which again, is truncated at the start, and runs along to the same extent as the top line of text. If space is limited, I would give priority to the branch rather than the path.

**Comment by steveward** [comment]
Feedback in #8856 from @marcosswald:

>The best solution would be if submodules would be properly supported, i.e. whenever a project with submodules is added, the submodules would be automatically added to Git Desktop e.g. the repo list could be kind of a tree view with submodules being sub items of each project.
>
>A simple and useful work around would be if you could edit the name of the repos in the list, e.g. in my case I copuld simply rename the repos:
>
>project
project/module
module

**Comment by vdms-mark** [comment]
I'm not sure if this is directly related, but there's some overlap at least.  I often clone the same repo multiple times, and just name the directory in the file system 01, 02, 03, etc.  I would love it if we could identify repos in the Desktop app based on the file system somehow.  For example, my file system...
```
c:\repos\myuser\myrepo01
c:\repos\myuser\myrepo02
c:\repos\myuser\myrepo03
```

In the Desktop app they all appear like this...
```
myuser\myrepo
myuser\myrepo
myuser\myrepo
```

**Comment by steveward** [comment]
@vdms-mark thanks for sharing your workflow. We're also tracking feedback about the ability to rename repositories in https://github.com/desktop/desktop/issues/7856, which may be helpful for your situation.

**Comment by rsodre** [comment]
3 years avoiding submodules... I just can't stand this mess anymore, I need to move on.
I love the simplicity of GitHub Desktop, but I'm looking for another git client **just because of this issue**.
At least let us give each repo a display name!

**Comment by corneliavarlam** [comment]
Hi Pasquale there is an update on the NHS login button

- [ ]   NHS login button text: 'Login with NHS number'

![Screen Shot 2020-09-10 at 9 02 31 PM](https://user-images.githubusercontent.com/36041372/92787251-3a381d00-f3a9-11ea-8175-9aee0803e63d.png)

**Comment by dpeter99** [comment]
I came across this issue while looking for submodule support, as it was not intuitive that you need to open them separately.

Looks like this has been on the back burner for 5y?? I didn't even know GitHub desktop has been a thing that long.
Let's hope this gets addressed now that 3.0 is out

**Comment by sergiou87** [comment]
Hi all! We added support for repository aliases a year ago (see #12000) and we believe that should be enough at least to differentiate them in the list of repositories.

We'll try to make it easier for users to navigate to a repo's submodules, but we're still discussing about it. Hopefully, aliases will at least alleviate this specific problem.
